devices/constants.go

```diff
@@ -38,9 +42,91 @@ const (
+
+	// FirmwareInstallInitializing indicates the device is performing init actions to install the update
+	// this covers the redfish states - 'starting', 'downloading'
+	// no action is required from the callers part in this state
+	FirmwareInstallInitializing = "initializing"
+
+	// FirmwareInstallQueued indicates the device has queued the update, but has not started the update task yet
+	// this covers the redfish states - 'pending', 'new'
+	// no action is required from the callers part in this state
+	FirmwareInstallQueued = "queued"
+
+	// FirmwareInstallRunner indicates the device is installing the update
+	// this covers the redfish states - 'running', 'stopping', 'cancelling'
+	// no action is required from the callers part in this state
+	FirmwareInstallRunning = "running"
+
+	// FirmwareInstallComplete indicates the device completed the firmware install
+	// this covers the redfish state - 'complete'
+	FirmwareInstallComplete = "complete"
+
+	// FirmwareInstallFailed indicates the firmware install failed
+	// this covers the redfish states - 'interrupted', 'killed', 'exception', 'cancelled', 'suspended'
+	FirmwareInstallFailed = "failed"
+
+	// FirmwareInstallPowerCycleHost indicates the firmware install requires a host power cycle
+	// this covers the dell redfish state - 'scheduled'
+	FirmwareInstallPowerCyleHost = "powercycle-host"
+
+	FirmwareInstallUnknown = "unknown"
+
+	// device BIOS/UEFI POST code bmclib identifiers
+	POSTStateBootINIT = "boot-init/pxe"
+	POSTStateUEFI     = "uefi"
+	POSTStateOS       = "grub/os"
+	POSTCodeUnknown   = "unknown"
+
+	// Generic component slugs
+	// Slugs are set on Device types to identify the type of component
+	SlugBackplaneExpander     = "Backplane Expander"
+	SlugChassis               = "Chassis"
+	SlugTPM                   = "TPM"
+	SlugGPU                   = "GPU"
+	SlugCPU                   = "CPU"
+	SlugPhysicalMem           = "PhysicalMemory"
+	SlugStorageController     = "StorageController"
+	SlugStorageControllers    = "StorageControllers"
+	SlugBMC                   = "BMC"
+	SlugBIOS                  = "BIOS"
+	SlugDrive                 = "Drive"
+	SlugDrives                = "Drives"
+	SlugDriveTypePCIeNVMEeSSD = "NVMe PCIe SSD"
+	SlugDriveTypeSATASSD      = "Sata SSD"
+	SlugDriveTypeSATAHDD      = "Sata HDD"
+	SlugNIC                   = "NIC"
+	SlugNICs                  = "NICs"
+	SlugPSU                   = "Power Supply"
+	SlugPSUs                  = "Power Supplies"
+	SlugCPLD                  = "CPLD"
+	SlugEnclosure             = "ENCLOSURE"
+	SlugUnknown               = "unknown"
 )
 
 // ListSupportedVendors  returns a list of supported vendors
 func ListSupportedVendors() []string {
 	return []string{HP, Dell, Supermicro}
 }
+
+// VendorFromProductName attempts to identify the vendor from the given productname
+func VendorFromProductName(productName string) string {
+	n := strings.ToLower(productName)
```

Comment:
Is Intel the only vendor that might use lowercase?

Reply:
thanks, yeah this needed to be fixed for various vendors, I've fixed it for the current vendors we have defined as constants.